lib/live_ex.ex

```diff
@@ -26,32 +29,43 @@ defmodule LiveX do
       @doc """
       Dispatch an Action with a `type` and an optional payload.
       """
-      def dispatch(type, payload \\ %{}, socket) do
-        event = %{
-          type: String.to_atom(type),
-          payload: payload
-        }
+      def dispatch(type, payload \\ nil, socket)
+
+      @spec dispatch(atom, any, socket) :: map
+      def dispatch(type, payload, socket) when is_atom(type) do
+        action = %{type: type, payload: payload}
 
-        send(socket.assigns.pid, event)
+        send(socket.assigns.pid, action)
+      end
+
+      @spec dispatch(String.t(), any, socket) :: map
+      def dispatch(type, payload, socket) do
+        type
+        |> String.to_atom()
```

Comment:
In case someone uses dynamic event names this might lead to exceeding the maximum number of allowed atoms within the VM. Atoms are also not garbage collected. I would recommend to stick with strings for the types.

Reply:
Good point. I changed from Atoms to Strings here